### PR TITLE
MAINT: Include optional license_family option in examples about section

### PR DESCRIPTION
**Pull request by MSeifert04**
See #1837 for the motivation for this PR.

### CODE REVIEW

File: recipes/example/meta.yaml [hunk 1]
```diff
@@ -45,6 +45,8 @@ about:
   # Prefer the short version, e.g: GPL-2.0 instead of GNU General Public License version 2.0
   # See https://opensource.org/licenses/alphabetical
   license: MIT
+  # If the license has different variants uncomment and edit the next line.
+  # license_family: BSD, Apache, GPL
```
Comment (MSeifert04) [inline recipes/example/meta.yaml]: I was not sure if the family be specified in any case. 

I removed the comment and included your suggestion. In theory `license_family` should be self-explaining anyways.

File: recipes/example/meta.yaml [hunk 2]
```diff
@@ -45,6 +45,7 @@ about:
   # Prefer the short version, e.g: GPL-2.0 instead of GNU General Public License version 2.0
   # See https://opensource.org/licenses/alphabetical
   license: MIT
+  license_family: MIT
```
Comment (jakirkham) [inline recipes/example/meta.yaml]: So I like the positioning of this, but it makes it seem required, which is not something we have agreed on. Could we maybe add a one line comment above the `license_family` that notes this is an optional field?

Comment (MSeifert04) [inline recipes/example/meta.yaml]: Sorry, I misunderstood https://github.com/conda-forge/staged-recipes/pull/1838#discussion_r85647035 . I included a comment, but I'm still not sure if it's correct. Is it always optional or just if it's identical to `license`?

**Comment by MSeifert04** [comment]
@jakirkham Any idea why the last commit triggered CircleCI?

**Comment by jakirkham** [comment]
It's a bug on CircleCI.

xref: https://discuss.circleci.com/t/ci-skip-does-not-work-in-pull-requests/5308

File: recipes/example/meta.yaml (hunk 2 above)
Comment (jakirkham) [inline recipes/example/meta.yaml]: No worries. `license_family` is optional, but encouraged. However, there has been a strong push a couple of times to make it required. So at least showing it in the `example` with an optional note is some level of compromise.

**Comment by MSeifert04** [comment]
@jakirkham Let me know as soon as you think this is ready or what should be reworded, so I can squash it.

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**